core/primitives/src/types.rs

```diff
@@ -173,6 +173,28 @@ pub struct RawStateChangesWithTrieKey {
     pub changes: Vec<RawStateChange>,
 }
 
+/// Consolidate state change of trie_key and the final value the trie key will be changed to
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
+pub struct ConsolidatedStateChange {
+    pub trie_key: TrieKey,
+    pub value: Option<Vec<u8>>,
+}
+
+pub type ConsolidatedStateChanges = Vec<ConsolidatedStateChange>;
+
+#[easy_ext::ext(ConsolidatedStateChangesExt)]
```

Comment:
tbh, not a big fan of `easy_ext`: it uses proc-macros *and* traits (both are rather advanced Rust features) to implement essentially a syntactical `foo(x) -> x.foo()` transformation. 

I'd just 

```rust
impl ConsolidatedStateChange {
  pub fn from_raw(changes: &[RawStateChangesWithTrieKey]) -> Vec<Self>
}
```